label_studio/io_storages/base_models.py

```diff
@@ -398,30 +423,37 @@ def _scan_and_create_links(self, link_class):
                     f'"Treat every bucket object as a source file"'
                 )
 
-            task = self.add_task(data, self.project, maximum_annotations, max_inner_id, self, key, link_class)
-            max_inner_id += 1
-
-            # update progress counters for storage info
-            tasks_created += 1
-
-            # add task to webhook list
-            tasks_for_webhook.append(task)
+            tasks_to_create.append(data)
+            keys.append(key)
+            if len(keys) == settings.CLOUD_IMPORT_TASK_CREATE_BATCH_SIZE:
+                created_tasks = self.add_tasks_batch(
+                    tasks_to_create, keys, self.project, maximum_annotations, max_inner_id, self, link_class
+                )
+                tasks_created += len(created_tasks)
+                self.info_update_progress(last_sync_count=tasks_created, tasks_existed=tasks_existed)
 
-            # settings.WEBHOOK_BATCH_SIZE
-            # `WEBHOOK_BATCH_SIZE` sets the maximum number of tasks sent in a single webhook call, ensuring manageable payload sizes.
-            # When `tasks_for_webhook` accumulates tasks equal to/exceeding `WEBHOOK_BATCH_SIZE`, they're sent in a webhook via
-            # `emit_webhooks_for_instance`, and `tasks_for_webhook` is cleared for new tasks.
-            # If tasks remain in `tasks_for_webhook` at process end (less than `WEBHOOK_BATCH_SIZE`), they're sent in a final webhook
-            # call to ensure all tasks are processed and no task is left unreported in the webhook.
-            if len(tasks_for_webhook) >= settings.WEBHOOK_BATCH_SIZE:
+                tasks_for_webhook.extend(created_tasks)
```

Comment:
Have you checked what WEBHOOK_BATCH_SIZE usually is, and if the batch size mismatch between this and CLOUD_IMPORT_TASK_BATCH_SIZE would cause any problems? Rn the comment says webhooks can be sent when num tasks >= this size, but the code will always send when num_tasks == this size, so I wonder if we depend on that anywhere. Also, the pathological case is something like, import batch size is 10000 and webhook batch size is 10, which'd significantly change behavior, so maybe check for import batch size > webhook batch size and logger.warning it?

Reply:
Its 100. To prevent it from getting too large. Now we will just use CLOUD_IMPORT_TASK_BATCH_SIZE which is 50, so not a problem. Dont foresee CLOUD_IMPORT_TASK_BATCH_SIZE ever getting too large here as we want to keep the size of the batch relatively small (the json tasks we are importing could be large in size)